flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/KubernetesScalingRealizer.java

```diff
@@ -43,6 +53,32 @@ public void realize(
                         getOverrideString(context, parallelismOverrides));
     }
 
+    @Override
+    public void realizeMemoryOverrides(
+            KubernetesJobAutoScalerContext context, Configuration configOverrides) {
+
+        if (context.getResource() instanceof FlinkDeployment) {
+            var flinkDeployment = ((FlinkDeployment) context.getResource());
+
+            flinkDeployment.getSpec().getFlinkConfiguration().putAll(configOverrides.toMap());
+
+            var totalMemoryOverride = MemoryTuningUtils.getTotalMemory(configOverrides, context);
+            if (totalMemoryOverride.compareTo(MemorySize.ZERO) > 0) {
+                Resource tmResource = flinkDeployment.getSpec().getTaskManager().getResource();
+                // Make sure to support the Kubernetes syntax here, which supports more formats than
+                // Flink's classes.
+                var currentMemory =
+                        new MemorySize(
+                                (Quantity.parse(tmResource.getMemory()).getNumericalAmount())
```

Comment:
Thanks for raising this. I intentionally added this way of parsing because I thought it is compatible with Kubernetes. Flink's MemorySize is a subset of Kubernetes because it does not support base 2 notation (e.g. Gigibytes). I didn't realize Flink is base 2 by default. I also forgot about the M vs MB notation. I'm afraid this logic might be broken in other parts of the operator. I'll look into it.

Reply:
Should be fixed now via https://github.com/apache/flink-kubernetes-operator/pull/762/commits/693e913122b197ed7dcc5d8d5b564f64d3b13168.